test/remote/gateways/remote_nmi_test.rb

```diff
@@ -388,6 +388,18 @@ def test_purchase_using_stored_credential_recurring_mit
     assert_success purchase
   end
 
+  def test_purchase_using_ntid_override_mit
+    initial_options = stored_credential_options(:merchant, :recurring, :initial)
+    assert purchase = @gateway.purchase(@amount, @credit_card, initial_options)
+    assert_success purchase
+    assert network_transaction_id = purchase.params['transactionid']
+
+    @options[:network_transaction_id] = purchase.params['transactionid']
```

Comment:
You could use the `network_transaction_id` variable you just set a couple lines above this 😁

Reply:
Thanks @jcreiff I agree with your suggestions and pushed the updates, thanks!